os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -405,127 +355,175 @@ static int binary_manager_reload(int bin_idx)
 #endif
 
 /****************************************************************************
- * Name: binary_manager_update
+ * Name: loading_thread
  *
  * Description:
- *   This function will terminate all the task/thread created by the binary
- *   i.e input binary name.
- *   If the binary is registered, it checks current running version and
- *   executes registered callback for unloading.
- *   And it terminates its children, unloads binary and loads the binary.
+ *   This thread loads binary with binary index.
  *
- * Input parameters:
- *   bin_name   -   The name of binary to be reload
+ ****************************************************************************/
+static int loading_thread(int argc, char *argv[])
+{
+	if (argc <= 1) {
+		bmdbg("Invalid arguments for loading, argc %d\n", argc);
+		return ERROR;
+	}
+
+	/* [1] binary index for loading */
+	return binary_manager_load((int)atoi(argv[1]));
+}
+
+/****************************************************************************
+ * Name: loadingall_thread
  *
- * Returned Value:
- *   None
+ * Description:
+ *   This function loads all user binaries in binary table.
```

Comment:
You mean `This function loads all user binaries WHICH ARE in binary table.` ? or `This function loads all user binaries FROM binary table.`?